### PR TITLE
fix: install vendored node_modules when using hardlinks

### PR DESCRIPTION
**Pull request by eemelipa**
### What does this PR do?

Fixes [issue](https://github.com/oven-sh/bun/issues/8093) where vendored `node_modules` aren't installed with hardlink backend.

Vendored `node_module` is a folder inside an installed package which is called `node_module`. So the library has bundled some of their dependencies into the published package instead of specifying it in their `package.json`. For example

An npm package is called `vendor-baz` and it contains the following files:

```
x index.js
x package.json
x cjs/
x cjs/node_modules/
x cjs/node_modules/foo-dep/
x cjs/node_modules/foo-dep/index.js
```

The contents of `cjs/node_modules/foo-dep/` is a vendored package.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [X] Code changes

### How did you verify your code works?

I wrote automated tests

- [ ] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [X] I included a test for the new code, or an existing test covers it
- [ ] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed
- [X] I wrote TypeScript/JavaScript tests and they pass locally (`bun-debug test test-file-name.test`)

### CODE REVIEW

File: test/regression/issue/08093.test.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  root_url,
+  setHandler,
+} from "./../../cli/install/dummy.registry.js";
+
+beforeAll(dummyBeforeAll);
+afterAll(dummyAfterAll);
+beforeEach(dummyBeforeEach);
+afterEach(dummyAfterEach);
+
+it("should install vendored node_modules with hardlink", async () => {
```
Comment (eemelipa) [inline test/regression/issue/08093.test.ts:24]: Similar testing approach as tests in `test/cli/install/bun-install.test.ts` and more specifically test case `should prefer latest-tagged dependency`

File: src/install/install.zig [hunk 2]
```diff
@@ -1366,7 +1366,7 @@ pub const PackageInstall = struct {
             cached_package_dir,
             this.allocator,
             &[_]bun.OSPathSlice{},
-            &[_]bun.OSPathSlice{bun.OSPathLiteral("node_modules")},
```
Comment (dylan-conway) [inline src/install/install.zig:1369]: TIL you can publish an npm package with a `node_modules` folder as long as it isn't top level

**Review by dylan-conway — APPROVED** [review]
Looks good to me. Will wait for CI to run

**Comment by dylan-conway** [comment]
We probably also want to do this for `installWithSymlink`, but it can be addressed in a future pr. I'm guessing there's a good chance top level `node_modules` will actually exist for local packages, so we would need to skip it

**Comment by dylan-conway** [comment]
Thank you

**Pull request merged**